src/main/java/org/example/service/ApartmentService.java

```diff
@@ -0,0 +1,45 @@
+package org.example.service;
+
+import org.example.dto.ApartmentDTO;
+import org.example.entity.Apartment;
+import org.example.repository.ApartmentRepository;
+import org.example.repository.RealtyAgentRepository;
+import org.example.util.TransformerDtoApartment;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class ApartmentService {
+
+    @Autowired
+    protected ApartmentRepository apartmentRepository;
+
+    @Autowired
+    protected RealtyAgentRepository realtyAgentRepository;
+
+    @Autowired
+    protected TransformerDtoApartment transformerDtoApartment;
+
+    public Optional<Apartment> findApartmentById(int id) {
```

Comment:
You shouldn't return Optional here - you should return Apartment, as it was before.